commands/azureCommands/pull-from-azure.ts

```diff
@@ -0,0 +1,56 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE.md in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Registry } from "azure-arm-containerregistry/lib/models";
+import { exec } from 'child_process';
+import { AzureImageTagNode, AzureRepositoryNode } from '../../explorer/models/azureRegistryNodes';
+import { ext } from '../../extensionVariables';
+import * as acrTools from '../../utils/Azure/acrTools';
+import { AzureImage } from "../../utils/Azure/models/image";
+import { Repository } from "../../utils/Azure/models/repository";
+import * as quickPicks from '../utils/quick-pick-azure';
+
+/* Pulls an image from Azure. The context is the image node the user has right clicked on */
+export async function pullFromAzure(context?: AzureImageTagNode | AzureRepositoryNode): Promise<any> {
+    let registryName: string;
+    let registry: Registry;
+    let imageName: string;
+
+    if (context) { // Right Click
+        registryName = context.registry.loginServer;
+        registry = context.registry;
+
+        if (context.contextValue === "azureImageTagNode") { // Right Click on AzureImageNode
+            imageName = context.label;
```

Comment:
Please don't use context.label to get the image name.  That's a problem with the original codebase (it's assuming the label will be the same as the programmatic image name) that I've been slowly correcting.  Please make the following changes/additions to AzureImageTagNode and then use the ImageNameWithTag property:

export class AzureImageTagNode extends NodeBase {
    constructor(
...
    ) {
CHANGED:
        super(AzureImageTagNode.getImageNameWithTag(repositoryName, tag));
    }

ADDED:
    public static getImageNameWithTag(repositoryName: string, tag: string): string {
        return `${repositoryName}:${tag}`
    }

    public get ImageNameWithTag(): string {
        return AzureImageTagNode.getImageNameWithTag(this.repositoryName, this.tag);
    }